src/dmd/cppmangle.d

```diff
@@ -524,15 +536,26 @@ private final class CppMangleVisitor : Visitor
         //printf("source_name(%s)\n", s.toChars());
         if (TemplateInstance ti = s.isTemplateInstance())
         {
-            if (!substitute(ti.tempdecl))
+            bool needsTa = false;
+            if (substitute(ti.tempdecl))
+                template_args(ti);
+            else if (this.writeStdSubstitution(ti, needsTa))
+            {
+                if (needsTa)
+                    template_args(ti);
+            }
+            else
             {
                 append(ti.tempdecl);
-                this.writeIdentifier(ti.tempdecl.toAlias().ident);
+                this.writeNamespace(
+                    s.namespace, () {
+                        this.writeIdentifier(ti.tempdecl.toAlias().ident);
+                        template_args(ti);
+                    });
```

Comment:
`source_name` isn't supposed to insert `N..E`  but `writeNamespace` does.
```d
extern(C++, "a", "b")
struct S
{
    void func();
}
pragma(msg, S.func.mangleof);
```
This is a regression. It prints `_ZNN1a1b1SE4funcEv` it used to be `_ZN1a1b1S4funcEv`.